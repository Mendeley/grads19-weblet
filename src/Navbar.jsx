import React from "react";
import styled from "styled-components";
import { Link, useHistory } from "react-router-dom";
import Button from "./Button";
import { logoutUser } from "./api";
import { withCookies, useCookies } from "react-cookie";
import { cookieOptions, cookieName } from "./Constants/Cookies";

export const StyledNavbar = styled.nav`
  background: #322d38;
  width: auto;
  height: 4vh;
  text-align: left;
`;

const StyledList = styled.ul`
  margin: 0;
  padding: 0;
  list-style-type: none;
`;

const StyledListItem = styled.li`
  display: inline-block;
  margin: 10px;
`;

export const StyledLink = styled(Link)`
  color: white;
`;

const Navbar = () => {
  const history = useHistory();
  const logout = async () => {
    try {
      await logoutUser(cookies.sessionToken.token);
      deleteSessionToken();
      history.push("/");
    } catch (error) {
      console.log(error);
    }
  };

  const [cookies, setCookie, removeCookie] = useCookies([cookieName]);
  const deleteSessionToken = () => {
    removeCookie(cookieName, cookieOptions);
  };

  return (
    <StyledNavbar>
      <StyledList>
        <StyledListItem>
          <StyledLink to="/">Home</StyledLink>
        </StyledListItem>
        {cookies.sessionToken ? (
          <>
            <StyledListItem>
              <StyledLink className="addConference" to="/add">
                Add Conference
              </StyledLink>
            </StyledListItem>
            <StyledListItem>
              <StyledLink
                className="profilePage"
                to={`/users/${cookies.sessionToken.userId}`}
              >
                Profile
              </StyledLink>
            </StyledListItem>
            <StyledListItem>
              <Button className="logout" navLink onClick={logout}>
                Logout
              </Button>
            </StyledListItem>
          </>
        ) : (
          <>
            <StyledListItem>
              <StyledLink className="register" to="/users/register">
                Register
              </StyledLink>
            </StyledListItem>
            <StyledListItem>
              <StyledLink className="login" to="/users/login">
                Login
              </StyledLink>
            </StyledListItem>
          </>
        )}
      </StyledList>
    </StyledNavbar>
  );
};

export default withCookies(Navbar);
