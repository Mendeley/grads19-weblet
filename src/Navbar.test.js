import Navbar from "./Navbar";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { NavLink } from 'react-router-dom';
import React from "react";

configure({ adapter: new Adapter() });
//what is adapter? Does it adapt enzyme for our version of react?

jest.mock('react-router-dom', () => {
    return {
        NavLink: () => "hey I'm a navlink"
        //defines a mock navlink for testing purposes
    }
});

afterEach(() => {
    jest.clearAllMocks()
    //clearing all the mocks we are creating after each test
});

describe("Navbar", () => {
    it("renders a navbar with text", () => {
        expect.assertions(2);

        const component = shallow(<Navbar />);

        expect(component).toMatchSnapshot();
        expect(component.find(NavLink).childAt(0).text()).toEqual('Home');
    });
});