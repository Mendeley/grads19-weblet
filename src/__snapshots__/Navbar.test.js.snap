// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders a navbar with text 1`] = `
<styled.nav>
  <Link
    to="/"
  >
    Home
  </Link>
</styled.nav>
`;
